src/features/user/infrastructure.rs

```diff
@@ -179,6 +186,58 @@ pub(crate) async fn save_vote_to_db(app_ctx: &AppContext, vote: Vote) -> Result<
     Ok(result.rows_affected())
 }
 
+/// Convenience function for getting categories by their snap ID, since it takes multiple API calls
+async fn get_categories_by_id(
```

Comment:
could call this `get_categories_from_snap_id` or similar?

Reply:
Its reasonable for a connivence function like this to reside in snapd-rs. Also, because this is kind of a 'lookup' something like `find_categories_by_snap_id` might be more intuitive.

Wdty?

Reply:
It does exist in `snapd-rs`, all it does is call the corresponding function and convert them to the protobuf Categories.

Reply:
`snapd_categories_by_snap_id` isn't my favorite but it does at least make it clear it makes a call out to `snapd`, was my thinking.